mmdeploy/codebase/mmdet/models/dense_heads/guided_anchor_head.py

```diff
@@ -0,0 +1,236 @@
+            (batch_size, num_priors * 1, H, W). Default None.
+        img_metas (list[dict]):  Meta information of the image, e.g.,
+            image size, scaling factor, etc.
+        cfg (mmcv.Config | None): Test / postprocessing configuration,
+            if None, test_cfg would be used. Default: None.
+        rescale (bool): If True, return boxes in original image space.
+            Default False.
+        with_nms (bool): If True, do nms before return boxes.
+            Default: True.
+    Returns:
+        If with_nms == True:
+            tuple[Tensor, Tensor]: tuple[Tensor, Tensor]: (dets, labels),
+            `dets` of shape [N, num_det, 5] and `labels` of shape
+            [N, num_det].
+        Else:
+            tuple[Tensor, Tensor, Tensor]: batch_mlvl_bboxes, batch_mlvl_scores
+    """
+    assert len(cls_scores) == len(bbox_preds) == len(shape_preds) == len(
+        loc_preds)
+    num_levels = len(cls_scores)
+    featmap_sizes = [featmap.size()[-2:] for featmap in cls_scores]
+    device = cls_scores[0].device
+    # get guided anchors
+
+    cfg = self.test_cfg if cfg is None else cfg
+    deploy_cfg = ctx.cfg
+    post_params = get_post_processing_params(deploy_cfg)
+    iou_threshold = cfg.nms.get('iou_threshold', post_params.iou_threshold)
+    score_threshold = cfg.get('score_thr', post_params.score_threshold)
+
+    _, guided_anchors = self.get_anchors(
+        featmap_sizes,
+        shape_preds,
+        loc_preds,
+        img_metas,
+        use_loc_filter=not self.training,
+        device=device)
+
+    result_proposals = []
+    result_labels = []
+    for img_id in range(len(img_metas)):
+        cls_score_list = [
+            cls_scores[i][img_id].detach() for i in range(num_levels)
+        ]
+        bbox_pred_list = [
+            bbox_preds[i][img_id].detach() for i in range(num_levels)
+        ]
+        guided_anchor_list = [
+            guided_anchors[img_id][i].detach() for i in range(num_levels)
+        ]
+
+        img_shape = img_metas[img_id]['img_shape']
+
+        mlvl_proposals = []
+        mlvl_labels = []
+        for idx in range(len(cls_score_list)):
+            rpn_cls_score = cls_score_list[idx]
+            rpn_bbox_pred = bbox_pred_list[idx]
+            anchors = guided_anchor_list[idx]
+            print(rpn_cls_score.shape, rpn_bbox_pred.shape)
+            assert rpn_cls_score.size()[-2:] == rpn_bbox_pred.size()[-2:]
+            # if no location is kept, end.
+
+            rpn_cls_score = rpn_cls_score.permute(1, 2, 0)
+            if self.use_sigmoid_cls:
+                rpn_cls_score = rpn_cls_score.reshape(-1)
+                scores = rpn_cls_score.sigmoid()
+            else:
+                rpn_cls_score = rpn_cls_score.reshape(-1, 2)
+                # remind that we set FG labels to [0, num_class-1]
+                # since mmdet v2.0
+                # BG cat_id: num_class
+                scores = rpn_cls_score.softmax(dim=1)[:, :-1]
```

Comment:
This logic seems to follow [`GARPNHead`](https://github.com/open-mmlab/mmdetection/blob/31c84958f54287a8be2b99cbf87a6dcf12e57753/mmdet/models/dense_heads/ga_rpn_head.py#L126-L138) instead of [`GuidedAnchorHead`](https://github.com/open-mmlab/mmdetection/blob/31c84958f54287a8be2b99cbf87a6dcf12e57753/mmdet/models/dense_heads/guided_anchor_head.py#L823-L826).

Is it expected behavior for rewriting `GuidedAnchorHead.get_bboxes`?

Reply:
I Fixed it.